akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala

```diff
@@ -0,0 +1,184 @@
+
+          expectResponse() shouldBe expectedResponse
+        }
+      }
+
+      "GET request in one HEADERS frame" in new SimpleRequestResponseRoundtripSetup {
+        requestResponseRoundtrip(
+          streamId = 1,
+          request = HttpRequest(uri = "http://www.example.com/"),
+          expectedHeaders = Seq(
+            ":method" -> "GET",
+            // TODO check if this makes sense?
+            ":scheme" -> "https",
+            ":authority" -> "www.example.com",
+            ":path" -> "/",
+            "content-length" -> "0"
+          ),
+          response = Seq(
+            // TODO shouldn't this produce an error since stream '1' is already the outgoing stream?
+            HeadersFrame(streamId = 1, endStream = true, endHeaders = true, HPackSpecExamples.C61FirstResponseWithHuffman, None)
+          ),
+          expectedResponse =
+            HPackSpecExamples.FirstResponse
+              .withEntity(Strict(ContentTypes.NoContentType, ByteString.empty))
+        )
+      }
+      "GOAWAY when the response has an invalid headers frame" in new TestSetup with NetProbes {
+        val requestStreamId = 0x1
+        emitRequest(requestStreamId, HttpRequest(uri = "http://www.example.com/"))
+        expectFrame() shouldBe a[HeadersFrame]
+
+        val responseStreamId = 0x2
+        val headerBlock = hex"00 00 01 01 05 00 00 00 01 40"
+        sendFrame(HeadersFrame(responseStreamId, endStream = true, endHeaders = true, headerBlock, None))
+
+        val (_, error) = expectGOAWAY(1)
+        error should ===(ErrorCode.COMPRESSION_ERROR)
+
+        // TODO we'd expect an error response here I think?
+      }
+      "GOAWAY when the response to a second request on different stream has an invalid headers frame" in new SimpleRequestResponseRoundtripSetup {
+        requestResponseRoundtrip(
+          streamId = 1,
+          request = HttpRequest(uri = "http://www.example.com/"),
+          expectedHeaders = Seq(
+            ":method" -> "GET",
+            // TODO check if this makes sense?
+            ":scheme" -> "https",
+            ":authority" -> "www.example.com",
+            ":path" -> "/",
+            "content-length" -> "0"
+          ),
+          response = Seq(
+            // TODO shouldn't this produce an error since stream '1' is already the outgoing stream?
+            HeadersFrame(streamId = 2, endStream = true, endHeaders = true, HPackSpecExamples.C61FirstResponseWithHuffman, None)
+          ),
+          expectedResponse =
+            HPackSpecExamples.FirstResponse
+              .withEntity(Strict(ContentTypes.NoContentType, ByteString.empty))
+        )
+
+        emitRequest(3, HttpRequest(uri = "http://www.example.com/"))
+        expectFrame() shouldBe a[HeadersFrame]
+
+        val incorrectHeaderBlock = hex"00 00 01 01 05 00 00 00 01 40"
+        sendHEADERS(4, endStream = true, endHeaders = true, headerBlockFragment = incorrectHeaderBlock)
+
+        // TODO shouldn't this be '3'?
```

Comment:
Ultimately, it should be but that flag is best effort and our best effort is currently quite low...